samples/teams-blazor-server/Pages/_Host.cshtml

```diff
@@ -0,0 +1,47 @@
+﻿@page "/"
+@namespace MGT_Teams_Blazor_Server.Pages
+@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
+@{
+    Layout = null;
+}
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>MGT_Teams_Blazor_Server</title>
+    <base href="~/" />
+    <link rel="stylesheet" href="css/bootstrap/bootstrap.min.css" />
+    <link href="css/site.css" rel="stylesheet" />
+
+    @* Including Teams SDK and Microsoft Grpah Toolkit Loader*@
+    <script src="https://unpkg.com/%40microsoft/teams-js/dist/MicrosoftTeams.min.js" crossorigin="anonymous"></script>
+    <script src="https://unpkg.com/%40microsoft/mgt/dist/bundle/mgt-loader.js"></script>
+    @*test script to automate the click on the sign-in button*@
+    @*<script>
+        window.addEventListener('load', (event) => {
+            document.querySelector("body > app > div > div > div:nth-child(3) > mgt-login").shadowRoot.querySelector("div > div > button").click();
+        });
+
+    </script>*@
```

Comment:
This is the part that I was looking for from this entire sample 😉 I'll be curious about @nmetulev feedback on this. Is there a better approach to get a reference to the `mgt-login` component and "force" a login without having to go through a shadow DOM click of a button? What if `mgt-login` had some methods to allow us to know that there is a token available, and that it could just leverage it and not require the click action?

Reply:
This is a good catch. 

To clarify, is the purpose of clicking the button here to initiate the sign in when the page loads? If so, you can accomplish this in a cleaner way by using the Providers service - for example `mgt.Providers.globalProvider.login()`. The `Providers` namespace also exposes a `handleProviderStateChanged` method you can use to get notified when the provider is initialized or when the state of the provider changes (signed in, signed out)

Reply:
Jep, purpose of that line was to create a "silent sign in" behaviour. But if there is a service for it, even better. Thank you!

Reply:
If you are not using this code, consider removing it.